pkg/apiserver/profiling/service.go

```diff
@@ -150,18 +152,21 @@ func (s *Service) exclusiveExecute(ctx context.Context, req *StartRequest) (*Tas
 }
 
 func (s *Service) startGroup(ctx context.Context, req *StartRequest) (*TaskGroup, error) {
-	taskGroup := NewTaskGroup(s.params.LocalStore, req.DurationSecs, model.NewRequestTargetStatisticsFromArray(&req.Targets))
+	taskGroup := NewTaskGroup(s.params.LocalStore, req.DurationSecs, model.NewRequestTargetStatisticsFromArray(&req.Targets), req.RequstedProfilingTypes)
 	if err := s.params.LocalStore.Create(taskGroup.TaskGroupModel).Error; err != nil {
 		log.Warn("failed to start task group", zap.Error(err))
 		return nil, err
 	}
 
 	tasks := make([]*Task, 0, len(req.Targets))
 	for _, target := range req.Targets {
-		t := NewTask(ctx, taskGroup, target, s.fetchers)
-		s.params.LocalStore.Create(t.TaskModel)
-		s.tasks.Store(t.ID, t)
-		tasks = append(tasks, t)
+		profileTypeList := req.RequstedProfilingTypes
+		for _, profilingType := range profileTypeList {
```

Comment:
better to check whether `profileTypeList` is a valid ProfilingType (since JSON deserialization will never check enum variants. Bad user may pass in "foo" as a profilingType). Otherwise bad request should be thrown.